make_tools.mk

```diff
@@ -0,0 +1,24 @@
+# This controls building executables in the `tools` folder.
+# Can be invoked through the `Makefile` or standalone.
+
+MAKEFLAGS += --no-print-directory
+
+# Inclusive list. If you don't want a tool to be built, don't add it here.
+# Use path relative to this makefile, in order fo Makefiles in subdirectory to work
+MKFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+TOOLS_DIR := $(dir $(MKFILE_PATH))/tools
+TOOL_NAMES := aif2pcm bin2c gbafix gbagfx mid2agb preproc scaninc
+
+TOOLDIRS := $(TOOL_NAMES:%=$(TOOLS_DIR)/%)
+
+# Tool making doesnt require a katam dependency scan.
+RULES_NO_SCAN += tools check-tools clean-tools $(TOOLDIRS)
+.PHONY: $(RULES_NO_SCAN)
```

Comment:
I don't really understand this part. For example, If I edit `tools/gbafix/elf.h`, the tool will not be rebuilt.

Reply:
Might be the specific constant you modified. If I change `ELFMAG` from `"\177ELF"` to `"\177ELe"` a rebuild does get triggered. I'm guessing not all constants on that header are used

Reply:
Welp somehow I triggered it before but it doesn't now so nvm. It's weird though, if I change `gbafix.c` itself it does rebuild. Must be an issue with gbafix's Makefile

Reply:
Right. It seems other tools are handling header dependencies in their own makefiles so they don't require scaninc (since there's only one non-phony target to be built). For some reason it's missing in gbafix's.

Reply:
Done! Ig that means pokeemerald has that bug too

Reply:
I guess so. You can submit a patch there. And probably another for fixing KEEP_TEMPS